### PR TITLE
RDK-34519 [DAC] Update BundleGen to support latest Dobby functionality

### PR DESCRIPTION
**Pull request by PriyaPesala**

### CODE REVIEW

**Review by TeknoVenus — CHANGES_REQUESTED** [review]
Please also rebase your changes so git history only contains the relevant commits

File: bundlegen/core/bundle_processor.py [hunk 1]
```diff
@@ -927,23 +931,54 @@ def _process_logging(self):
...
+                logging_plugin = {
+                    "required": True,
+                    "data": {
+                        "sink": "file",
+                        "fileOptions": {
+                            "path": logfile,
+                        }
+                    }
+                }
+                self._add_annotation('run.oci.hooks.stderr','/dev/stderr')
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Added in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 2]
```diff
@@ -927,23 +931,54 @@ def _process_logging(self):
...
+                self._add_annotation('run.oci.hooks.stdout','/dev/stdout')
+            else:
+                limit = self.platform_cfg.get('logging').get("limit")
+                logging_plugin = {
+                    "required": True,
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Removed in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 3]
```diff
@@ -927,23 +931,54 @@ def _process_logging(self):
...
+                        "sink": "journald"
+                    }
+                }
+            else:
+                priority = self.platform_cfg['logging']['journaldOptions']['priority']
+                logging_plugin = {
+                    "required": True,
+                    "data": {
+                        "sink": "journald",
+                        "journaldOptions": {
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Removed in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 4]
```diff
@@ -1057,3 +1092,74 @@ def _process_seccomp(self):
         self.oci_config['linux']['seccomp'] = {}
         self.oci_config['linux']['seccomp'] = self.platform_cfg.get('seccomp')
 
+    def _process_ipc(self):
+        """
+        Adds ipc plugin inside rdkPlugins
+        """
+        if self.app_metadata['ipc']['enable']:
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Updated in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 5]
```diff
@@ -1057,3 +1092,74 @@ def _process_seccomp(self):
...
+                    "required": True,
+                    "data": plugin
+                    }
+                self.oci_config['rdkPlugins']['ipc'] = ipc_plugin
+        return
+    def _process_minidump(self):
+        """
+        Adds midump plugin inside rdkPlugins
+        """
+        if self.app_metadata['minidump']['enable']:
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Updated in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 6]
```diff
@@ -1057,3 +1092,74 @@ def _process_seccomp(self):
...
+                    "data": plugin
+                    }
+                self.oci_config['rdkPlugins']['minidump'] = minidump_plugin
+        return
+
+    def _process_oomcrash(self):
+        """
+        Adds oomcrash plugin inside rdkPlugins
+        """
+        if self.app_metadata['oomcrash']['enable']:
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Updated in latest patchset

File: bundlegen/core/bundle_processor.py [hunk 7]
```diff
@@ -1057,3 +1092,74 @@ def _process_seccomp(self):
...
+        return
+
+    def _process_gpu_plugin(self):
+        """
+        Adds GPU plugin inside rdkPlugins
+        """
+        if self.app_metadata["resources"].get('gpu'):
+            app_gpu_requirement = self.app_metadata.get('resources').get('gpu')
+            app_gpu_bytes = humanfriendly.parse_size(app_gpu_requirement, binary=True)
+            plugin = app_gpu_bytes
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/core/bundle_processor.py]: Removed this variable and updated latest patchset

File: bundlegen/schema/appMetadataSchema.json [hunk 8]
```diff
@@ -31,7 +32,33 @@
...
+                "enable": {"type":"boolean"}
+            }
+        },
+        "minidump": {
+            "type": "object",
+            "properties": {
+                "enable": {"type":"boolean"}
+            }
+        },
+        "oomcrush": {
```
[the remark this answers is not in the document]
Comment (PriyaPesala) [reply bundlegen/schema/appMetadataSchema.json]: Modified in latest patchset

**Comment by PriyaPesala** [comment]
> Please also rebase your changes so git history only contains the relevant commits

In latest patchset changes are rebased

**Review by TeknoVenus — APPROVED** [review]

**Pull request merged**